[Streams] Add a test for piping to errored

ricea opened the pull request:
The test 'Piping from an errored readable stream to an errored writable
stream' was misnamed, as the destination writable was erroring, not
errored. Rename it. Add a variant of the test where pipeTo() isn't
called until the destination is errored.

ricea committed:
[Streams] Add a test for piping to errored

The test 'Piping from an errored readable stream to an errored writable
stream' was misnamed, as the destination writable was erroring, not
errored. Rename it. Add a variant of the test where pipeTo() isn't
called until the destination is errored.

domenic commented on streams/piping/multiple-propagation.js:
```diff
@@ -35,6 +35,32 @@ promise_test(t => {
+  });
+
+  // Trying to abort a stream that was errored will give that error back
+  return Promise.resolve()
```
The test might be more obviously correct if you checked the errored state of `ws` before doing the piping, instead of checking it at the end.

ricea replied:
Done. I also split a couple more tests into erroring / errored variants, although in these cases the asserts don't change. PTAL.

domenic reviewed (approved):
LGTM with potential improvement.

ricea committed:
Explicitly wait for writable to be errored + more

Explicitly wait for the writable to be errored. This is clearer than
implicitly waiting via Promise.resolve().

Also split more tests into erroring + errored.

ricea committed:
Factor out CreateErroredWritableStream(t) function

domenic reviewed (approved):
Very nice. Will let you squash and merge, or rebase and merge, as desired.
